packages/sdk/src/Portfolio/Integrations/MysoV3.ts

```diff
@@ -0,0 +1,396 @@
+}
+
+export function withdrawTokensFromEscrowsDecode(encoded: Hex): WithdrawTokensFromEscrowsArgs {
+  const [{ escrows, tokens }] = decodeAbiParameters(withdrawTokensFromEscrowsEncoding, encoded);
+  return { escrows, tokens };
+}
+
+//------------------------------------------------------------------------------
+// SWEEP
+//------------------------------------------------------------------------------
+
+const sweepEncoding = [
+  {
+    type: "tuple",
+    name: "sweepArgs",
+    components: [
+      {
+        type: "address[]",
+        name: "tokens",
+      },
+    ],
+  },
+] as const;
+
+export function sweepEncode(args: SweepArgs): Hex {
+  return encodeAbiParameters(sweepEncoding, [args]);
+}
+
+export function sweepDecode(encoded: Hex): SweepArgs {
+  const [{ tokens }] = decodeAbiParameters(sweepEncoding, encoded);
+  return { tokens } as const;
+}
+
+//------------------------------------------------------------------------------
+// ACTION FUNCTIONS
+//------------------------------------------------------------------------------
+export const createEscrowByTakingQuote = ExternalPositionManager.makeUse(
+  Action.CreateEscrowByTakingQuote,
+  createEscrowByTakingQuoteEncode,
+);
+
+export const createEscrowByStartingAuction = ExternalPositionManager.makeUse(
+  Action.CreateEscrowByStartingAuction,
+  createEscrowByStartingAuctionEncode,
+);
+
+export const closeAndSweepEscrows = ExternalPositionManager.makeUse(
+  Action.CloseAndSweepEscrows,
+  closeAndSweepEscrowsEncode,
+);
+
+export const withdrawTokensFromEscrows = ExternalPositionManager.makeUse(
+  Action.WithdrawTokensFromEscrows,
+  withdrawTokensFromEscrowsEncode,
+);
+
+export const sweep = ExternalPositionManager.makeUse(Action.Sweep, sweepEncode);
+
+//------------------------------------------------------------------------------
+// EXTERNAL READ FUNCTIONS
+//------------------------------------------------------------------------------
+export function getManagedAssets(client: Client, args: Viem.ContractCallParameters<{ position: Address }>) {
+  return readContract(client, {
+    ...Viem.extractBlockParameters(args),
+    abi: parseAbi(["function getManagedAssets() view returns (address[] assets_, uint256[] amounts_)"]),
+    functionName: "getManagedAssets",
+    address: args.position,
+    args: [],
+  });
+}
+
+export function getDebtAssets(client: Client, args: Viem.ContractCallParameters<{ position: Address }>) {
+  return readContract(client, {
+    ...Viem.extractBlockParameters(args),
+    abi: parseAbi(["function getDebtAssets() view returns (address[] assets_, uint256[] amounts_)"]),
+    functionName: "getDebtAssets",
+    address: args.position,
+    args: [],
+  });
+}
+
+export function getNumOpenEscrows(client: Client, args: Viem.ContractCallParameters<{ position: Address }>) {
+  return readContract(client, {
+    ...Viem.extractBlockParameters(args),
+    abi: parseAbi(["function getNumOpenEscrows() view returns (uint256 numOpenEscrows_)"]),
```

Comment:
We usually add at least lib to the abis package. Then you can use this abi here

Reply:
Nice, didn't know about that package. refactored to use that instead.